### PR TITLE
Fixed Issue 22594 - Update "Interfacing to C" to include intptr_t and uintptr_t

### PR DESCRIPTION
**Pull request by ioanavivi12**
I added an extra row to include intptr_t and uintptr_t data types.

### CODE REVIEW

**Review by maxhaton — CHANGES_REQUESTED** [review]

File: spec/interfaceToC.dd [hunk 1]
```diff
@@ -152,6 +152,7 @@ $(H2 $(LNAME2 data_type_compat, Data Type Compatibility))
 
         $(TROW3 core.stdc.config.c_long, long, long)
         $(TROW3 core.stdc.config.c_ulong, unsigned long, unsigned long)
+        $(TROW3 core.stdc.stdint, intptr_t, uintptr_t)
```
[the remark this answers is not in the document]
Comment (ioanavivi12) [reply spec/interfaceToC.dd]: I fixed it now. I hope it is ok.

File: spec/interfaceToC.dd [hunk 2]
```diff
@@ -152,6 +152,7 @@ $(H2 $(LNAME2 data_type_compat, Data Type Compatibility))
 
         $(TROW3 core.stdc.config.c_long, long, long)
         $(TROW3 core.stdc.config.c_ulong, unsigned long, unsigned long)
+        $(TROW2 core.stdc.stdint.intptr_t, intptr_t, $(D uintptr_t))
```
Comment (RazvanN7) [inline spec/interfaceToC.dd:155]: You are putting both on the same row, but what is actually needed is:

```
 $(TROW2 core.stdc.stdint.intptr_t, intptr_t, intptr_t)
 $(TROW2 core.stdc.stdint.uintptr_t, uintptr_t, uintptr_t)
```

**Review by RazvanN7 — CHANGES_REQUESTED** [review]

File: spec/interfaceToC.dd (hunk 2 above)
Comment (RazvanN7) [reply spec/interfaceToC.dd:155]: you still need to delete this row.

Comment (ioanavivi12) [reply spec/interfaceToC.dd]: I did it, thank you. I hope it is ok now.

File: spec/interfaceToC.dd [hunk 3]
```diff
@@ -152,9 +152,10 @@ $(H2 $(LNAME2 data_type_compat, Data Type Compatibility))
 
         $(TROW3 core.stdc.config.c_long, long, long)
         $(TROW3 core.stdc.config.c_ulong, unsigned long, unsigned long)
+        $(TROW3 core.stdc.stdint.intptr_t, intptr_t, intptr_t)
+        $(TROW3 core.stdc.stdint.uintptr_t, uintptr_t, uintptr_t)
 
         $(TROW3PLUS long, long long, long, (or $(D long long)))
-        $(TROW3PLUS ulong, unsigned long long, unsigned long, (or $(D unsigned long long)))
```
Comment (RazvanN7) [inline spec/interfaceToC.dd:157]: Why is this deleted?

File: spec/interfaceToC.dd [hunk 4]
```diff
@@ -412,3 +414,4 @@ Macros:
     TROW3PLUS=$(TR3 $(TD $(D $1)), $(TD $(D $2)), $(TD $(D $3) $4))
         TD2=$(MULTICOL_CELL 2, $0)
         TITLE=Interfacing to C
+
```
Comment (RazvanN7) [inline spec/interfaceToC.dd:417]: This change is unrelated to this PR.

**Comment by RazvanN7** [comment]
Please squash commits.

**Comment by Geod24** [comment]
Please include the issue name as part of the commit title and PR title.
For people watching this repository, "Fix issue 22594" has no meaning, and they would like to know if they need to open the notification or not.
